subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4468,22 +4453,15 @@ static int32_t lwm2m_engine_service(const int64_t timestamp)
 
 int lwm2m_engine_context_close(struct lwm2m_ctx *client_ctx)
 {
-	struct observe_node *obs, *tmp;
-	sys_snode_t *prev_node = NULL;
+	struct observe_node *obs;
 	int sock_fd = client_ctx->sock_fd;
 	struct lwm2m_message *msg;
 	size_t i;
 
 	/* Remove observes for this context */
-	SYS_SLIST_FOR_EACH_CONTAINER_SAFE(&engine_observer_list, obs, tmp,
+	SYS_SLIST_FOR_EACH_CONTAINER(&client_ctx->observer, obs,
 					  node) {
-		if (obs->ctx == client_ctx) {
-			sys_slist_remove(&engine_observer_list, prev_node,
```

Comment:
I'd keep the explicit `sys_slist_remove` instead of assuming that memset is enough.